### PR TITLE
Mccalluc/testing search

### PR DESCRIPTION
**Pull request by mccalluc**
@john-conroy : If you're more familiar with jest mocking than me, and can get Search.spec.js unstuck, great! But time box it, and if it's not happening, let's comment out the new failing test, and get just one new test in?

### CODE REVIEW

File: context/app/static/js/pages/search/Search.spec.js [hunk 1]
```diff
@@ -4,8 +4,33 @@ import { render } from 'test-utils/functions';
...
 import Search from './Search';
 
-test('Search error if no URL param', () => {
+test('Search error if missing entity_type param', () => {
+  // TODO: I want to supress the log noise during this test... but this doesn't work.
+  global.console = { error: jest.fn().mockImplementation(() => {}) };
+
   expect(() => render(<Search elasticsearchEndpoint="not-used" title="not-used" />)).toThrow(
     'Unexpected URL param "entity_type[0]="; Should be one of {donor, sample, dataset}',
   );
```
Comment (john-conroy) [inline context/app/static/js/pages/search/Search.spec.js:13]:
```suggestion
  const originalConsoleError = console.error;
  console.error = jest.fn();

  expect(() => render(<Search elasticsearchEndpoint="not-used" title="not-used" />)).toThrow(
    'Unexpected URL param "entity_type[0]="; Should be one of {donor, sample, dataset}',
  );

  console.error = originalConsoleError;
```

File: context/app/static/js/pages/search/Search.spec.js [hunk 2]
```diff
@@ -4,8 +4,33 @@ import { render } from 'test-utils/functions';
...
+  const originalWindow = { ...window };
+  const windowSpy = jest.spyOn(global, 'window', 'get');
+  windowSpy.mockImplementation(() => ({
+    ...originalWindow,
+    location: {
+      ...originalWindow.location,
+      search: '?entity_type[0]=Donor',
+    },
+  }));
+
```
Comment (john-conroy) [inline context/app/static/js/pages/search/Search.spec.js:27]:
```suggestion
  const location = {
    ...window.location,
    search: '?entity_type[0]=Donor',
  };
  Object.defineProperty(window, 'location', {
    writable: true,
    value: location,
  });

```

**Review by john-conroy — COMMENTED** [review]
You also need to change `const { searchParams } = new URL(document.location);` to `const searchParams = new URLSearchParams(window.location.search);` in Search.

**Comment by mccalluc** [comment]
(The one error from the previous build is something we've seen earlier: `gyp ERR! stack Error: not found: make`)

File: context/app/static/js/pages/search/Search.spec.js [hunk 3]
```diff
@@ -1,11 +1,41 @@
...
+    'Donor Metadata',
+    'Affiliation',
+    '0 results found',
+    'Age',
+    'BMI',
+    'Sex',
+    'Race',
+    'Last Modified',
+  ];
+  expectedStrings.forEach((s) => expect(screen.getByText(s)));
```
Comment (john-conroy) [inline context/app/static/js/pages/search/Search.spec.js:40]:
```suggestion
  expectedStrings.forEach((s) => expect(screen.getByText(s)).toBeInTheDocument());
```
Not a big deal, but I prefer to make assertions explicit as noted [here by the react-testing-library creator](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions).

**Review by john-conroy — APPROVED** [review]
Looks good, thanks!

**Pull request merged**